mission_control/test/test_if_mission_control_aborts_when_health_monitor_reports_fault.py

```diff
@@ -0,0 +1,141 @@
+
+
+def wait_for(predicate, period=1):
+    while not rospy.is_shutdown():
+        result = predicate()
+        if result:
+            return result
+        rospy.sleep(period)
+    return predicate()
+
+
+class TestMissionControlAbortsWhenHealthMonitorReportsFault(unittest.TestCase):
+    """ 
+        Simulate error message sent by health monitor and checks 
+        if the mission control publishes the abort status.
+        The steps int the involved are:
+            1)  Load a mission
+            2)  Execute the Mission
+            3)  Simulate an error sent by health monitor
+            4)  check if the mission control abort the mission
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        rospy.init_node('test_mission_control_aborts_procedure')
+
+    def setUp(self):
+        self.dir_path = os.path.dirname(
+            os.path.abspath(__file__)) + '/test_files/'
+        self.mission_to_load = LoadMission()
+        self.mission_to_load.mission_file_full_path = self.dir_path + \
+            "test_missions/mission.xml"
+        self.mission_to_execute = ExecuteMission()
+        self.mission_to_execute.mission_id = 1
+        self.simulate_error_code = ReportFault()
+        self.simulate_error_code.fault_id = ReportFault.PAYLOAD_ERROR
+        self.mission_load_state = ReportLoadMissionState()
+        self.report_execute_mission = ReportExecuteMissionState()
+
+        # Subscribers
+        self.exec_state_sub = rospy.Subscriber('/mission_control_node/report_mission_execute_state',
+                                               ReportExecuteMissionState, self.callback_mission_execute_state)
+
+        self.load_state_sub = rospy.Subscriber('/mission_control_node/report_mission_load_state',
+                                               ReportLoadMissionState, self.callback_mission_load_state)
+
+        #   Publisher
+
+        self.simulated_mission_control_load_mission_pub = rospy.Publisher('/mission_control_node/load_mission',
+                                                                          LoadMission, latch=True, queue_size=1)
+
+        self.simulated_mission_control_execute_mission_pub = rospy.Publisher(
+            '/mission_control_node/execute_mission', ExecuteMission, latch=True, queue_size=1)
+
+        self.simulated_health_monitor_pub = rospy.Publisher('/health_monitor/report_fault',
+                                                            ReportFault, queue_size=1)
+
+    def callback_mission_execute_state(self, msg):
+        self.report_execute_mission = msg
+
+    def callback_mission_load_state(self, msg):
+        self.mission_load_state = msg.load_state
+        self.mission_load_state_flag = True
+
+    def test_mission_control_aborts_if_health_monitor_reports_fault(self):
+
+        # Load Mission
+        self.simulated_mission_control_load_mission_pub.publish(
+            self.mission_to_load)
+
+        def test_load_valid_mission():
```

Comment:
@patapufate nit: remove the `test_` prefix in wait predicates. It'll read better.

Reply:
Fixed